Annotato/Annotato/Persistence/PersistenceManager/OnlinePersistenceService.swift

```diff
@@ -0,0 +1,71 @@
+import AnnotatoSharedLibrary
+import Foundation
+
+struct OnlinePersistenceService: PersistenceService {
+    private let remotePersistence: PersistenceManager
+    private let localPersistence: PersistenceManager
+
+    init(remotePersistence: PersistenceManager, localPersistence: PersistenceManager) {
+        self.remotePersistence = remotePersistence
+        self.localPersistence = localPersistence
+    }
+
+    func getOwnDocuments(userId: String) async -> [Document]? {
+        await remotePersistence.documents.getOwnDocuments(userId: userId)
+    }
+
+    func getSharedDocuments(userId: String) async -> [Document]? {
+        await remotePersistence.documents.getSharedDocuments(userId: userId)
+    }
+
+    func getDocument(documentId: UUID) async -> Document? {
+        await remotePersistence.documents.getDocument(documentId: documentId)
+    }
+
+    func createDocument(document: Document) async -> Document? {
+        guard let createdDocumentRemote = await remotePersistence
+            .documents
+            .createDocument(document: document) else {
+            return nil
+        }
+        guard let createdDocumentLocal = await localPersistence
+            .documents
+            .createDocument(document: createdDocumentRemote) else {
+            return nil
+        }
+        return createdDocumentLocal
+    }
+
+    func updateDocument(document: Document) async -> Document? {
+        guard let updatedDocumentRemote = await remotePersistence.documents.updateDocument(document: document) else {
+            return nil
+        }
+        guard let updatedDocumentLocal = await localPersistence
+            .documents
+            .updateDocument(document: updatedDocumentRemote) else {
+            return nil
+        }
+        return updatedDocumentLocal
+    }
+
+    func deleteDocument(document: Document) async -> Document? {
+        guard let deletedDocumentRemote = await remotePersistence.documents.deleteDocument(document: document) else {
+            return nil
+        }
+        guard let deletedDocumentLocal = await localPersistence
+            .documents
+            .deleteDocument(document: deletedDocumentRemote) else {
+            return nil
+        }
+        return deletedDocumentLocal
+    }
+
+    func createDocumentShare(documentShare: DocumentShare) async -> DocumentShare? {
```

Comment:
Let's separate the methods related to `DocumentsPersistence` and `DocumentSharesPersistence`. You can define them as extensions in separate files. For example, in `OnlinePersistenceService+DocumentSharesPersistence`,
```swift
extension OnlinePersistenceService: DocumentSharesPersistence {
    func createDocumentShare(...) {
        ...
    }
}
```

Reply:
Ok